cmd/gcp-controller-manager/node_annotator.go

```diff
@@ -286,6 +308,54 @@ func extractKubeLabels(instance *compute.Instance) (map[string]string, error) {
 	return parsedLabels, nil
 }
 
+func extractNodeTaints(instance *compute.Instance) ([]core.Taint, error) {
+	const kubeEnvKey = "kube-env"
+
+	if instance.Metadata == nil {
+		return nil, errNoMetadata
+	}
+
+	var kubeEnv *string
+	for _, item := range instance.Metadata.Items {
+		if item == nil || item.Key != kubeEnvKey {
+			continue
+		}
+		if item.Value == nil {
+			return nil, fmt.Errorf("instance %q had nil %q", instance.SelfLink, kubeEnvKey)
+		}
+		kubeEnv = item.Value
+		break
+	}
+	if kubeEnv == nil {
+		return nil, errNoMetadata
+	}
+	if len(*kubeEnv) == 0 {
+		klog.Infof("Node taints not found in instance metadata, %s is empty", kubeEnvKey)
+		return nil, nil
+	}
+
+	var taintsEnv string
+	for _, env := range strings.Split(*kubeEnv, ";") {
+		if strings.HasPrefix(env, "node_taints") {
+			taintsEnv = env
```

Comment:
nit: we can break the loop here as well.

Reply:
Done